### PR TITLE
Fix call to fetch subscriptions over gRPC but not implemented.

### PR DESCRIPTION
**Pull request by artursouza**
Fix call to fetch subscriptions over gRPC but not implemented.

This issue is detected in the Java SDK and `git bisect` accused commit eeca309f0ca1630d85415cd4a6c4d52cbaaf4092 as the root cause. This change will handle `not implemented` case correctly in gRPC.

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/pubsub/subscriptions.go [hunk 1]
```diff
@@ -199,7 +199,8 @@ func GetSubscriptionsGRPC(channel runtimev1pb.AppCallbackClient, log logger.Logg
 				s, ok := status.FromError(rErr)
 				if ok && s != nil {
 					if s.Code() == codes.Unimplemented {
-						return rResp, backoff.Permanent(rErr)
+						log.Infof("pubsub subscriptions: gRPC app does not implement ListTopicSubscriptions")
```
Comment (mukundansundar) [inline pkg/runtime/pubsub/subscriptions.go:202]:
```suggestion
						log.Warnf("pubsub subscriptions: gRPC app does not implement ListTopicSubscriptions")
```

File: pkg/runtime/pubsub/subscriptions.go [hunk 2]
```diff
@@ -214,7 +215,8 @@ func GetSubscriptionsGRPC(channel runtimev1pb.AppCallbackClient, log logger.Logg
 			if rErr != nil {
 				s, ok := status.FromError(rErr)
 				if ok && s != nil && s.Code() == codes.Unimplemented {
-					return rResp, backoff.Permanent(rErr)
+					log.Infof("pubsub subscriptions: gRPC app does not implement ListTopicSubscriptions")
```
Comment (mukundansundar) [inline pkg/runtime/pubsub/subscriptions.go:218]:
```suggestion
					log.Warnf("pubsub subscriptions: gRPC app does not implement ListTopicSubscriptions")
```

**Review by mukundansundar — COMMENTED** [review]
Can a unit test be added for this?

**Comment by artursouza** [comment]
> Can a unit test be added for this?

Well. There is one as I see but the expectations are wrong on that one. So I will fix it.

**Review by ItalyPaleAle — APPROVED** [review]
LGTM but agree with Mukundan’s feedback that log message may be a warning in this case. If people use the Dapr SDKs, it should be implemented for them anyways.

File: pkg/runtime/pubsub/subscriptions.go [hunk 3]
```diff
@@ -199,7 +199,8 @@ func GetSubscriptionsGRPC(channel runtimev1pb.AppCallbackClient, log logger.Logg
 				s, ok := status.FromError(rErr)
 				if ok && s != nil {
 					if s.Code() == codes.Unimplemented {
-						return rResp, backoff.Permanent(rErr)
+						log.Infof("pubsub subscriptions: gRPC app does not implement ListTopicSubscriptions")
+						return rResp, nil
```
Comment (ItalyPaleAle) [inline pkg/runtime/pubsub/subscriptions.go:203]: What is rResp in this case? `nil`? Should we just return `nil` (or another zero value) here?

**Review by ItalyPaleAle — DISMISSED** [review]

File: pkg/runtime/pubsub/subscriptions.go (hunk 1 above)
Comment (artursouza) [reply pkg/runtime/pubsub/subscriptions.go:202]: Why warning? Apps are not required or expected to implement this. It means a hello world app would get this warning - which is why I am using Info.

File: pkg/runtime/pubsub/subscriptions.go (hunk 3 above)
Comment (artursouza) [reply pkg/runtime/pubsub/subscriptions.go]: It is not nil, but I am changing this and the existing code to guarantee that and not depend on this behavior from the downstream function.

File: pkg/runtime/pubsub/subscriptions.go (hunk 2 above)
Comment (artursouza) [reply pkg/runtime/pubsub/subscriptions.go:216]: I believe this should be Info since it is really not expected for apps to implement this.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**